### PR TITLE
Downsample reads with ReservoirDownsampler in CNNScoreVariants.

### PR DESCRIPTION
**Pull request by cmnbroad**

### CODE REVIEW

**Comment by droazen** [comment]
@lucidtronix Hopefully by the time you see this the tests will have finished and you can update them if the output looks good. If it's not looking good, just let me know and we can wait until after the 4.1 release.

**Comment by lucidtronix** [comment]
A single site changed by a small amount in CNN score which was expected. (We weren't 100% sure we were testing any sites where downsampling would kick in, so this is good from a test coverage perspective.  I think this is safe to include.   I'm updating the test file and will push once tests pass locally.

**Comment by droazen** [comment]
Great, thanks @lucidtronix ! On a related note, was https://github.com/broadinstitute/gatk/pull/5251 supposed to have been merged for 4.1?

**Comment by lucidtronix** [comment]
I think #5251 can wait.

**Comment by lucidtronix** [comment]
So... the reservoir downsampler seems to be sensitive to the state of the random number generator.  Since several tests depend on the same expected output file I keep having failures. I'm resetting the seed for each test now to see if that will work.

**Comment by droazen** [comment]
@lucidtronix Try adding calls to `Utils.resetRandomGenerator()` at the start of each test case. We do this in `HaplotypeCallerIntegrationTest` as well, for the same reason.

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariantsIntegrationTest.java [hunk 1]
```diff
@@ -155,6 +156,9 @@ public void testOnContigEdge() throws IOException {
      */
     @Test(groups = {"python"})
     public void testInference2dResourceModel() throws IOException {
+        // We reset the random number generator at the beginning of each test so that the random down-sampling of reads
+        // by the reservoir down-sampler does not cause slightly different scores.
+        Utils.getRandomGenerator().setSeed(1234);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariantsIntegrationTest.java]: Call `Utils.resetRandomGenerator()` instead (here and below)

Comment (lucidtronix) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariantsIntegrationTest.java]: got it!

**Pull request merged**